### ISSUE TITLE
Emergency rules support

### ISSUE THREAD

**Issue by maniekq** [aetas/mqgateway#5]
In case there is a problem with MQTT broker/connection there should be emergency rules list for Gateway to be able to run simple tasks like turn on light on button press.

This should be implemented using REST API to be able to communicate between multiple gateways and avoid need for MQTT broker on Gateway device. Because of that this task depends on #4 .

It may be worth to look at the [Node-RED](https://nodered.org/) project for implementing rules.

**Comment by jozala** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Emergency rules support](https://trello.com/c/TyHBA0O8/27-emergency-rules-support)

**Comment by mattthhdp** [comment]
I don't think you should in store any of those. All of the logic should be done on Home assistant, openhab etc etc...
Because it's easier to make automation if you flip a switch you don't necessarily want to turn on or off on the one light.
I think the best approach should be to Ping the mqtt broker and send an email sms or something like that, and don't register any news button press. That way if the server come back at 2 am the light won't turn on.

**Comment by jozala** [comment]
Hi @mattthhdp,
Thanks for your comment and suggestion.
I think I may have not been clear about what I want to achieve in this task and "emergency rules support" feature in general. I have now edited the description of the task to make it more clear. 

In general, the use case behind this task is following:
I am not at home and Home Assistant/Openhab or MQTT broker has failed. I want the other family members to be still able to control basic elements of home automation system (lights are crucial in my opinion). So when MqGateway will find out about the problem (e.g. by receiving [LWT](https://www.hivemq.com/blog/mqtt-essentials-part-9-last-will-and-testament/) message on MQTT) it will enable emergency rules.

Emergency rules will support only very basic rules. I agree with you in general that all the logic should be done on the Home Assistant / OpenHab which are great in this area. It is not the point of this task to replace the automations in these systems. It is only about increasing the resilience.

Sending notification or email when MqGateway connection to MQTT broker fail is already possible. MqGateway has [LWT](https://www.hivemq.com/blog/mqtt-essentials-part-9-last-will-and-testament/) set and publish it's state to Home Assistant.

**Comment by mattthhdp** [comment]
I understand your concern, but still thing that it's not the job of this device (I also understand that with a conditional statements it should be pretty easy to activate or deactivate) in General if the mqtt broker  or the logic server is down. You will have a bunch of other problem and I don't thing that light will bother you that much.
For example, if the dht22 or BME sensor for temperature aren't working, no heating or cooling in the house.

**Comment by jozala** [comment]
Well, I believe it depends :) . In my case most of smart devices have another way of controlling them, different than with MqGateway. For example, heating control is possible manually on smart thermostatic valves and gas heater. Garage gate can be opened with remote control, etc. 
When part of the home automation system is down, it is still possible to control those elements. I am just loosing the "automation" part. With the lights it is different. I haven't found a way to be able to control them manually when home automation system is down. That is why I am looking for a solution to make this specific part more resilient.
Anyway, I have other elements of the system prepared for failure (like Home Assistant and MQTT broker failover on separate computer), so this is probably not the highest priority feature of MqGateway for me, but I would really like to implement it eventually.
